File: src/frontend/src/app/components/cadastro/setor/model/setor-dto.ts
export class SetorDTO {
  id?: string;
  nome: string;
  descricao?: string;
  centroCustoId: string;
  centroCustoNome?: string;
  createdAt?: string;
  updatedAt?: string;
  createdBy?: string;
  updatedBy?: string;

  constructor(
    id: string | undefined,
    nome: string,
    centroCustoId: string,
    descricao?: string,
    centroCustoNome?: string,
    createdAt?: string,
    updatedAt?: string,
    createdBy?: string,
    updatedBy?: string
  ) {
    this.id = id;
    this.nome = nome;
    this.descricao = descricao;
    this.centroCustoId = centroCustoId;
    this.centroCustoNome = centroCustoNome;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.createdBy = createdBy;
    this.updatedBy = updatedBy;
  }
}

File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/cadastro/setor/SetorDTO.java
package br.com.grupopipa.gestaointegrada.cadastro.setor;

import java.time.LocalDateTime;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SetorDTO implements DTO {
  private UUID id;
  private String nome;
  private String descricao;
  private UUID centroCustoId;
  private String centroCustoNome;
  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;
  private String createdBy;
  private String updatedBy;
}
